src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp

```diff
@@ -81,28 +86,155 @@ void VulnerabilityScannerFacade::start(
+            {
+                while (!stopRetryThread.load())
+                {
+                    for (int i = 0; i < SLEEP_RETRY_THREADS_SEC; ++i)
+                    {
+                        std::this_thread::sleep_for(std::chrono::seconds(1));
+                        if (stopRetryThread.load())
+                        {
+                            break;
+                        }
+                    }
+
+                    while (!syncQueue->empty() && !stopRetryThread.load())
+                    {
+                        std::shared_ptr<std::string> returnStr = syncQueue->pop();
+
+                        if (nullptr != returnStr)
+                        {
+                            flatbuffers::Verifier syncVerifier(reinterpret_cast<const uint8_t*>(returnStr->c_str()),
+                                                               returnStr->size());
+
+                            if (SyscollectorSynchronization::VerifySyncMsgBuffer(syncVerifier))
+                            {
+                                std::variant<const SyscollectorDeltas::Delta*,
+                                             const SyscollectorSynchronization::SyncMsg*>
+                                    data = SyscollectorSynchronization::GetSyncMsg(
+                                        reinterpret_cast<const char*>(returnStr->c_str()));
+
+                                try
+                                {
+                                    logDebug1(WM_VULNSCAN_LOGTAG, "Retry syscollector_packages event.");
+                                    scanOrchestrator->run(data);
+                                }
+                                catch (const std::exception& e)
+                                {
+                                    logWarn(WM_VULNSCAN_LOGTAG, "Discarded syscollector_packages event: %s", e.what());
+                                }
+                            }
+                        }
+                    }
+                }
+            });
+
         // Subscription to syscollector delta events.
         m_syscollectorDeltasSubscription =
             std::make_unique<RouterSubscriber>("deltas-syscollector", "vulnerability_scanner_deltas");
         m_syscollectorDeltasSubscription->subscribe(
-            [scanOrchestrator](const std::vector<char>& message)
+            [scanOrchestrator, deltaQueue = m_deltaQueue](const std::vector<char>& message)
             {
                 std::variant<const SyscollectorDeltas::Delta*, const SyscollectorSynchronization::SyncMsg*> data =
                     SyscollectorDeltas::GetDelta(message.data());
 
-                scanOrchestrator->run(data);
+                try
+                {
+                    scanOrchestrator->run(data);
+                }
+                catch (const OsDataException& e)
+                {
+                    deltaQueue->push(std::move(std::string(message.data(), message.size())));
```

Comment:
This is an error, the message is a binary payload, we can have a 0x0 in that payload and it would be taken as the end of a string when cast to this.

Extend the queue to save a slice.

Reply:
Using slice to push element